hub/api/dataset_utils.py

```diff
@@ -0,0 +1,39 @@
+import numcodecs
+from codecs import PngCodec, Lz4, NumPy, Zstd, JpegCodec
+
+
+def _get_compressor(compressor: str, **kwargs):
+    """Get compressor object
+
+    Example:
+        compressor = _get_compressor('lz4', acceleration=2)
+
+    Args:
+        compressor (str): lowercase name of the required compressor
+
+    Returns:
+        Codec object providing compression
+
+    Raises:
+        ValueError: if the name of compressor isn't in ['lz4', 'zstd', 'numpy', 'png', 'jpeg']
+    """
+    if compressor is None:
+        return None
+    elif compressor.lower() == "lz4":
+        acceleration = kwargs.get("acceleration", numcodecs.lz4.DEFAULT_ACCELERATION)
+        return Lz4(acceleration)
+    elif compressor.lower() == "zstd":
+        level = kwargs.get("level", numcodecs.zstd.DEFAULT_CLEVEL)
+        return Zstd(level)
+    elif compressor.lower() == "numpy":
+        return NumPy()
+    elif compressor.lower() == "png":
+        single_channel = kwargs.get("single_channel", True)
```

Comment:
Why is this the default? Am I misunderstanding the nature of this argument, or is it saying that the default is for grayscale images?

Reply:
It's just saying to check if image has 1 in shape and remove that channel. So even with single_channel = True images can have 3 channels.